app/code/Magento/Customer/Model/CustomerRegistry.php

```diff
@@ -77,8 +92,9 @@ public function retrieve($customerId)
         if (isset($this->customerRegistryById[$customerId])) {
             return $this->customerRegistryById[$customerId];
         }
-        /** @var Customer $customer */
-        $customer = $this->customerFactory->create()->load($customerId);
+        $customer = $this->customerFactory->create();
+        $this->customerResourceFactory->create()
```

Comment:
Resource model shouldn't contain any data inside, so it seems to me that we need to get customer resource model, not factory of it.
Is there any reason why you used factory there?